arch/riscv/Kconfig

```diff
@@ -383,6 +376,14 @@ config GEN_IRQ_VECTOR_TABLE
 config ARCH_HAS_SINGLE_THREAD_SUPPORT
 	default y if !SMP
 
+config ARCH_SUPPORTS_STACKWALK
```

Comment:
why define this per architecture? shouldn't this be in arch/Kconfig and then select it if arch supports it instead?

Reply:
Because each arch seems to have different dependencies

- riscv can be done with or without fp
- x86_32 requires fp, et al
- x86_64 requires fp, `NO_OPTIMIZATIONS`, et al
- arm64 requires fp

this per-arch Kconfig indicates that the dependencies are met and stackwalk can be enabled

But I realize that `ARCH_SUPPORTS_STACKWALK` is redundant, I could have (re)used the `ARCH_HAS_STACKWALK` for the dependency checking

Reply:
> But I realize that `ARCH_SUPPORTS_STACKWALK` is redundant, I could have (re)used the `ARCH_HAS_STACKWALK` for the dependency checking, patch incoming

done